src/scriptworker/cot/verify.py

```diff
@@ -1907,7 +1923,7 @@ async def trace_back_to_tree(chain):
     # check for restricted scopes.
     my_repo = repos[chain]
     for scope in chain.task["scopes"]:
-        if scope in rules["scopes"]:
+        if chain.is_scope_in_restricted_scopes(scope, rules["scopes"]):
             log.info("Found privileged scope {}".format(scope))
             level = rules["scopes"][scope]
```

Comment:
Hm. I think this line is going to break, since `scope` may not be the key; it may be a glob.
We may need to change `is_scope_in_restricted_scopes` to return the matching rule, which means we need to update the test as well.